File: src/main/java/org/xi/quick/codegenerator/properties/FileProperties.java
package org.xi.quick.codegenerator.properties;

import org.springframework.boot.context.properties.ConfigurationProperties;
import org.springframework.stereotype.Component;

import java.util.Set;

@Component
@ConfigurationProperties(prefix = "generator.file")
public class FileProperties {

    /**
     * 忽略的文件集合
     */
    private Set<String> ignore;

    /**
     * 聚合文件集合
     */
    private Set<String> aggregate;

    /**
     * 仅复制文件集合
     */
    private Set<String> justCopy;

    public Set<String> getIgnore() {
        return ignore;
    }

    public void setIgnore(Set<String> ignore) {
        this.ignore = ignore;
    }

    public Set<String> getAggregate() {
        return aggregate;
    }

    public void setAggregate(Set<String> aggregate) {
        this.aggregate = aggregate;
    }

    public Set<String> getJustCopy() {
        return justCopy;
    }

    public void setJustCopy(Set<String> justCopy) {
        this.justCopy = justCopy;
    }
}

File: code-generator-kt/src/main/kotlin/org/xi/quick/codegeneratorkt/properties/FileProperties.kt
package org.xi.quick.codegeneratorkt.properties

import org.springframework.boot.context.properties.ConfigurationProperties
import org.springframework.stereotype.Component
import java.util.HashSet

@Component
@ConfigurationProperties(prefix = "generator.file")
class FileProperties {

    /**
     * 忽略的文件集合
     */
    var ignore: Set<String>? = HashSet()

    /**
     * 聚合文件集合
     */
    var aggregate: Set<String>? = HashSet()

    /**
     * 仅复制文件集合
     */
    var justCopy: Set<String>? = HashSet()
}
